src/routes/dashboard/index.tsx

```diff
@@ -144,85 +130,50 @@ export default function Settings({ params }: Props) {
               <Action as={showAdvanced ? ChevronUp : ChevronDown} />
             </div>
           </AdvancedWrapper>
-          {showAdvanced &&
+
+          {(showAdvanced || searchInput.state) &&
             advancedSettings
               .filter(filterSearchResults)
               .map((setting, i) => (
                 <SettingListItem
                   displayName={setting.displayName}
                   description={setting.description}
                   icon={setting.icon}
-                  active={activeSetting === setting.name}
-                  onClick={() => setLocation("/" + setting.name)}
+                  active={activeSettingParam === setting.name}
+                  onClick={() =>
+                    navigate(`/${setting.name}` as DashboardRoutePath)
+                  }
                   key={`advanced-settings-${i}`}
                 />
               ))}
         </SettingsList>
       </Panel>
+
       <Panel normalPadding showRightBorder>
         <Spacer y={0.45} />
+
         <MidSettingsTitle>
-          {allSettings &&
-            browser.i18n.getMessage(
-              allSettings.find((s) => s.name === activeSetting)?.displayName ||
-                ""
-            )}
+          {browser.i18n.getMessage(actualActiveSetting?.displayName || "")}
         </MidSettingsTitle>
+
         <Spacer y={0.85} />
-        {activeSetting &&
-          ((definedSetting && (
-            <SettingEl
-              setting={settings.find((s) => s.name === activeSetting)}
-              key={activeSetting}
-            />
-          )) ||
-            (() => {
-              const Component = allSettings.find(
-                (s) => s.name === activeSetting
-              )?.component;
-
-              if (!Component) {
-                return <></>;
-              }
-
-              return <Component />;
-            })())}
-      </Panel>
-      <Panel>
-        {!!activeAppSetting && (
-          <AppSettings
-            app={activeAppSetting}
-            showTitle
-            key={activeAppSetting.url}
+
+        {isDashboardRouteConfig(actualActiveSetting) ? (
```

Comment:
`<Routes>` could (and should) also be used here. See https://github.com/arconnectio/ArConnect/issues/566